src/components/duration.rs

```diff
@@ -568,21 +568,23 @@ impl Duration {
         };
         // 35. Let calendarRec be ? CreateCalendarMethodsRecordFromRelativeTo(plainRelativeTo, zonedRelativeTo, « DATE-ADD, DATE-UNTIL »).
 
-        // TODO: Validate below expect -> Potentially return error.
-        let relative_to_date = relative_to.date.expect("Date must exist at this point.");
```

Comment:
suggestion: This can actually just be:

```rust

let relative_to_date = relative_to.date;

```

You should be able to just provide `relative_to_date` as an arg instead of `Some(relative_to_date)`